### PR TITLE
[rel/17.0] Update Newtonsoft.Json to 13.0.1 (#4309

### PR DESCRIPTION
**Pull request by nohwnd**
Update newtonsoft to 13.0.1

### CODE REVIEW

File: test/Microsoft.TestPlatform.AcceptanceTests/MultitargetingTestHostTests.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ public class MultitargetingTestHostTests : AcceptanceTestBase
     {
         [TestMethod]
         [TestCategory("Windows-Review")]
+        [Ignore("Unreliable")]
```
Comment (nohwnd) [inline test/Microsoft.TestPlatform.AcceptanceTests/MultitargetingTestHostTests.cs:17]: Ignore acceptance test for multi targetting, we did not touch that, it relies on an edge case of some api not being available across different versions of .net. It does not depend on newtonsoft and I don't know why it is broken.

**Review by Evangelink — APPROVED** [review]
